front: use Date instead of string in OsrdConfState.startTime

emersion opened the pull request:
_See individual commits._

Improves type safety.

emersion committed:
front: use Date instead of string in computeOpSchedules()

Improves type safety.

While at it, fix a small typo: s/substractTo/subtractFrom/

Signed-off-by: Simon Ser <contact@emersion.fr>

emersion committed:
front: use Date instead of string in extractDateAndTimefromISO()

Improves type safety.

Signed-off-by: Simon Ser <contact@emersion.fr>

emersion committed:
front: use Date instead of string in OsrdConfState.startTime

Improves type safety.

Previously, it was important for OsrdConfState.startTime to be in
a specific timezone, because the string was later truncated and
directly used on an datetime-local form input. This patch improves
robustness by converting to the local timezone right before feeding
the value to the datetime-local form input, making it so a different
timezone in OsrdConfState.startTime doesn't break the app.

IsoDateTimeString is not longer used and can be dropped.

Signed-off-by: Simon Ser <contact@emersion.fr>

clarani commented on front/src/reducers/osrdconf/operationalStudiesConf/index.ts:
```diff
@@ -34,7 +33,7 @@ export const operationalStudiesConfSlice = createSlice({
       state.rollingStockID = rollingStock?.id;
       state.pathSteps = computeBasePathSteps(action.payload);
-      state.startTime = convertIsoUtcToLocalTime(start_time);
+      state.startTime = new Date(start_time);
```
Instead of retransforming the start_time here, you can directly use startTime (which is already a Date ;) )

emersion replied:
I was worried about changing behavior in a refactoring, but it seems like this should be safe. Done!

clarani reviewed (commented):
Nice cleaning !! Thanks !

Found a bug: 
- when editing a train, the date is not displayed

RomainValls reviewed (approved):
LGTM (except for the bug mentioned by @clarani), tested ✅

emersion committed:
front: omit start_time from TrainScheduleWithDetails

startTime should be used instead, because it's an already-parsed
Date object instead of a string.

Signed-off-by: Simon Ser <contact@emersion.fr>

emersion commented:
Ooops, sounds like `formatLocalDateTime()` was using a bad format string. Fixed:

```diff
 export const formatLocalDateTime = (date: Date) =>
-  dayjs(date).local().format('YYYY-MM-DDTHH:MM:SS');
+  dayjs(date).local().format('YYYY-MM-DDTHH:mm:ss');
```

clarani reviewed (approved):
LGTM, tested :white_check_mark:
